### DISCUSSION TITLE
Allow "inheriting" (record primary) constructor parameters

### DISCUSSION THREAD

**Discussion by tmat** [dotnet/csharplang#6061] [Language Ideas]
Consider a record `A` with many properties and a record `B` that derives from `A` and adds some more.

Currently we need to write:

```C#
record class A(int X, int Y, int Z);
record class B(int X, int Y, int Z, int U, int V) : A(X, Y, Z);
```

Suppose now there are not 3 properties but a couple dozen or more. Then repeating all the parameters multiple times gets very noisy.

I propose that instead a new syntax is introduces that threads all the parameters through together without repeating their types and/or names, something like so:

```C#
record class A(int X, int Y, int Z);
record class B(base, int U, int V) : A(base);
```

If any parameters of `A` are optional then all extra parameters in B must be optional as well.

The `base` refers to all parameters of the primary constructor. This could be generalized to non-record types as long as the base type has a single constructor.

```C#
class A
{
   public A(int x, int y, int z) { ... } 
}

class B
{
  public B(base, int u, int v) : base(base) { }
}
```

**Comment by HaloFour** [comment]
> Suppose now there are not 3 properties but a couple dozen or more.

I'm not opposed to constructor inheritance in general, but positional construction of a "couple dozen or more" parameters sounds like a horrific pattern to want to encourage in the language.

Also: #806

**Comment by tmat** [comment]
>  "couple dozen or more" parameters sounds like a horrific pattern to want to encourage in the language.

It's what records are intended to be used for, no? 

Like so: https://sourceroslyn.io/#Microsoft.CodeAnalysis.CSharp.Workspaces/CSharpSyntaxFormattingOptions.cs,72

Not a couple dozen yet but potentially many parameters.

**Reply by CyrusNajmabadi** [reply]
Options are like the canonical case here :)

**Reply by HaloFour** [reply]
Sounds like a good use case for nominal records.  For more than maybe 3-4 parameters the benefits of positional construction and deconstruction is entirely lost.

**Comment by CyrusNajmabadi** [comment]
Note: in general i like this (or maybe syntax like `..base`) as a general pattern of just not having to repeat yourself.  It would be a pity if this was only for records/primary-constructors though, as i would love this just for any type that had a single constructor.  unfortunately, in that case, adding a constructor would be a source breaking change.

**Reply by HaloFour** [reply]
I agree, I'm not arguing against the general idea, just this use case.  😁

**Comment by TahirAhmadov** [comment]
This is a good idea in regular classes, too; typical use case for me is, business logic classes with multiple dependencies, and as I inherit those, I have to write out huge constructors which pass back all those dependencies (and potentially set a few dependencies on the child class itself). I proposed something similar at https://github.com/dotnet/csharplang/discussions/5601.
